vignettes/customizing-module-output.Rmd

```diff
@@ -0,0 +1,326 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` provides a collection of ready-to-use modules, each with a predefined way of presenting outputs.
+
+While these built-in modules cover many common use cases, you may need to customize their outputs to better fit your specific requirements.
+This document outlines the available customization options for modifying `teal` module outputs.
+
+You will learn how to use `teal_transform_module()` to modify and enhance the objects created by `teal::modules()`,
+allowing you to tailor the outputs without rewriting the original module code.
+
+## Decorators
+
+In `teal`, decorators are used to modify module outputs, such as plots or tables, without changing their core structure.
+
+For example, imagine you have a plot and you want to add a title.
+A decorator function, like `add_title(x, "My Plot")`, takes an existing plot and updates it by adding a title while keeping everything else the same.
+This makes it different from a function that creates a brand-new plot from scratch, as decorators work by enhancing what’s already there.
+
+The decorators process can vary in complexity:
+
+- **Simple Decorators**: Single-step modifications, such as a single method call that does not require additional data.
+- **Complex Decorators**: Multi-step operations that may involve interdependent transformations, potentially requiring input from dedicated `shiny` UI elements.
+
+This functionality empowers application developers to significantly customize outputs beyond the default capabilities provided by existing module parameters.
+Decorators allow for advanced modifications, enabling highly tailored and dynamic user experiences in `teal` applications.
```

Comment:
I don't understand what these two sentences means. They don't help app developer grasp what is decoration. Besides these statements doesn't have to be true because:

> This functionality empowers application developers to significantly customize outputs beyond the default capabilities provided by existing module parameters

Signification is debatable. For example one can't change object type from plot to for example table. Also decorator is "existing module parameter" so it doesn't go beyond "existing module parameters" - it goes beyond default output appearance.

Reply:
These two sentences feel like closing statements to me. 

However, since we already have this sentence at the beginning of the section:
```
In `teal`, decorators are used to modify module outputs, such as plots or tables, without changing their core structure.
```  

I think we can remove these two sentences without losing clarity.

WDYT?

Reply:
Yes, I think also to remove.